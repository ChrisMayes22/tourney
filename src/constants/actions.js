export const INITIALIZE_CHARACTER = 'INITIALIZE_CHARACTER';
export const REMOVE_CHARACTER = 'REMOVE_CHARACTER';
export const REMOVE_LOWEST_SCORE = 'REMOVE_LOWEST_SCORE';
export const RATE_CHARACTER = 'RATE_CHARACTER';
export const SUBMIT_CHARACTER_RATINGS = 'SUBMIT_CHARACTER_RATINGS';
export const SET_PLAYERS = 'SET_PLAYERS';
export const RESET_FOR_FINALS = 'RESET_FOR_FINALS';
export const CHOOSE_WINNER = 'CHOOSE_WINNER';

export const initializeCharacter = function(imageUrl, id){
    return {type: INITIALIZE_CHARACTER, 
        payload: {image: imageUrl, id}}
}
export const removeCharacter = function(id){
    console.log('Remove Character Dispatched')
    console.log(REMOVE_CHARACTER)
    return {type: REMOVE_CHARACTER, id}
}
export const removeLowestScore = function(){
    return {type: REMOVE_LOWEST_SCORE};
}
export const rateCharacter = function(points, rowId, character){
    return {type: RATE_CHARACTER, payload:{
        points, rowId, character}}
}
export const submitCharacterRatings = function(playersArray){
    return {type: SUBMIT_CHARACTER_RATINGS, payload:{
        players: playersArray
    }};
};
export const setPlayers = function(num){
    return {type: SET_PLAYERS, number: num}
};
export const resetForFinals = function(){
    return {type: RESET_FOR_FINALS}
}
export const chooseWinner = function(character){
    return {type: CHOOSE_WINNER, character}
}

