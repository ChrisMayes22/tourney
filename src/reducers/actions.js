export const INITIALIZE_CHARACTER = 'INITIALIZE_CHARACTER';
export const REMOVE_CHARACTER = 'REMOVE_CHARACTER';
export const HALVE_CHARACTERS = 'HALVE_CHARACTERS';
export const SELECT_VOTE_PAGE_CHARACTERS = 'SELECT_VOTE_PAGE_CHARACTERS';
export const RATE_CHARACTER = 'RATE_CHARACTER';
export const MATCH_CHARACTERS_AND_ROWS = 'MATCH_CHARACTERS_AND_ROWS';
export const SUBMIT_CHARACTER_RATINGS = 'SUBMIT_CHARACTER_RATINGS';