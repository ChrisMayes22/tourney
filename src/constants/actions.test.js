import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import configureStore from 'redux-mock-store';
import * as actions from './actions';

const mockStore = configureStore();
const store = mockStore();

describe('When actions are dispatched', function(){
    beforeEach(function(){
        store.clearActions();
    });
    describe('When initializeCharacter is dispatched', function(){
        test('dispatches the correct action and payload', function(){
            let action = [{type: actions.INITIALIZE_CHARACTER, 
                payload: {image: 'foo$url', id: 'foo$id'}}];

            store.dispatch(actions.initializeCharacter('foo$url', 'foo$id'));
            expect(store.getActions()).toEqual(action);
            expect(store.getActions()).toMatchSnapshot();
        })
    });
    describe('When removeCharacter is dispatched', function(){
        test('dispatches the correct action and payload', function(){
            let action = [{type: actions.REMOVE_CHARACTER, id: 'foo$id'}];

            store.dispatch(actions.removeCharacter('foo$id'));
            expect(store.getActions()).toEqual(action);
            expect(store.getActions()).toMatchSnapshot();
        })
    });
    describe('When removeLowestScore is dispatched', function(){
        test('dispatches the correct action and payload', function(){
            let action = [{type: actions.REMOVE_LOWEST_SCORE}];

            store.dispatch(actions.removeLowestScore());
            expect(store.getActions()).toEqual(action);
            expect(store.getActions()).toMatchSnapshot();
        })
    });
    describe('When rateCharacter is dispatched', function(){
        test('dispatches the correct action and payload', function(){
            let action = [{type: actions.RATE_CHARACTER, payload:{
                points: 1,
                rowId: 'foo$id',
                character: 'foo$character'
            }}];

            store.dispatch(actions.rateCharacter(1, 'foo$id', 'foo$character'));
            expect(store.getActions()).toEqual(action);
            expect(store.getActions()).toMatchSnapshot();
        })
    });
    describe('When submitCharacterRatings is dispatched', function(){
        test('dispatches the correct action and payload', function(){
            let action = [{type: actions.SUBMIT_CHARACTER_RATINGS, payload:{
                players: 'foo$players'
            }}];

            store.dispatch(actions.submitCharacterRatings('foo$players'));
            expect(store.getActions()).toEqual(action);
            expect(store.getActions()).toMatchSnapshot();
        })
    });
    describe('When setPlayers is dispatched', function(){
        test('dispatches the correct action and payload', function(){
            let action = [{type: actions.SET_PLAYERS, number: 3}];

            store.dispatch(actions.setPlayers(3));
            expect(store.getActions()).toEqual(action);
            expect(store.getActions()).toMatchSnapshot();
        })
    });
    describe('When resetForFinals is dispatched', function(){
        test('dispatches the correct action and payload', function(){
            let action = [{type: actions.RESET_FOR_FINALS}];

            store.dispatch(actions.resetForFinals());
            expect(store.getActions()).toEqual(action);
            expect(store.getActions()).toMatchSnapshot();
        })
    });
})
