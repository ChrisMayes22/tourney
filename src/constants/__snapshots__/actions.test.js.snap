// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When actions are dispatched When initializeCharacter is dispatched dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "id": "foo$id",
      "image": "foo$url",
    },
    "type": "INITIALIZE_CHARACTER",
  },
]
`;

exports[`When actions are dispatched When rateCharacter is dispatched dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "character": "foo$character",
      "points": 1,
      "rowId": "foo$id",
    },
    "type": "RATE_CHARACTER",
  },
]
`;

exports[`When actions are dispatched When removeCharacter is dispatched dispatches the correct action and payload 1`] = `
Array [
  Object {
    "id": "foo$id",
    "type": "REMOVE_CHARACTER",
  },
]
`;

exports[`When actions are dispatched When removeLowestScore is dispatched dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "REMOVE_LOWEST_SCORE",
  },
]
`;

exports[`When actions are dispatched When resetForFinals is dispatched dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "RESET_FOR_FINALS",
  },
]
`;

exports[`When actions are dispatched When setPlayers is dispatched dispatches the correct action and payload 1`] = `
Array [
  Object {
    "number": 3,
    "type": "SET_PLAYERS",
  },
]
`;

exports[`When actions are dispatched When submitCharacterRatings is dispatched dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "players": "foo$players",
    },
    "type": "SUBMIT_CHARACTER_RATINGS",
  },
]
`;
